api/utils/route_test.go

```diff
@@ -244,3 +244,71 @@ func TestRouteToServer(t *testing.T) {
 		})
 	}
 }
+
+// TestSSHRouteMatcherScoring verifies the expected scoring behavior of SSHRouteMatcher.
+func TestSSHRouteMatcherScoring(t *testing.T) {
```

Comment:
Shoudln't we have a test case for two nodes with the same IP and different ports?